eng/build.ps1

```diff
@@ -361,6 +361,9 @@ function TestUsingRunTests() {
       # Minimize all windows to avoid interference during integration test runs
       $shell = New-Object -ComObject "Shell.Application"
       $shell.MinimizeAll()
+
+      # Copy the correct (17.0) version of Microsoft.VisualStudio.Threading to the expected location
+      Copy-Item "C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\PublicAssemblies\Microsoft.VisualStudio.Threading.16.0\Microsoft.VisualStudio.Threading.dll" -Destination "C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\PrivateAssemblies" -Force
```

Comment:
@AArnott an unusable 15.8 build of Microsoft.VisualStudio.Threading is in the PrivateAssemblies folder, and the required 17.0 build is in a weird 16.0 subfolder. 😕

Reply:
Your Copy-Item step here looks bad. The 15.8 version under PrivateAssemblies is by design (some other processes expect to find a 15.8 assembly versioned assembly there. That's why for 16.x we added a new file with the incremented assembly version, and "hid" it in a directory that isn't in a typical probing path so that other processes can't lock us into not updating it going forward.
What tempted you to copy the new one over the older one? 

As for the now obsolete `.16.0` suffix on the folder name, we fix that for 17.0 in this PR: https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/336955